src/moonlink_connectors/src/rest_ingest/decimal_utils.rs

```diff
@@ -0,0 +1,238 @@
+                actual_precision,
+            } => {
+                assert_eq!(value, precision_exceeding_value.to_string());
+                assert_eq!(expected_precision, precision);
+                assert_eq!(actual_precision, 7);
+            }
+            _ => panic!("Expected a PrecisionExceeds error, but got a different variant: {err:?}"),
+        }
+
+        let scale_exceeding_value = "123.4567";
+        let precision = 8;
+        let scale = 3;
+        let err =
+            convert_decimal_to_row_value(scale_exceeding_value, precision, scale).unwrap_err();
+        match err {
+            DecimalConversionError::ScaleExceeds {
+                value,
+                expected_scale,
+                actual_scale,
+            } => {
+                assert_eq!(value, scale_exceeding_value.to_string());
+                assert_eq!(expected_scale, scale);
+                assert_eq!(actual_scale, 4);
+            }
+            _ => panic!("Expected a ScaleExceeds error, but got a different variant: {err:?}"),
+        }
+
+        let negative_scale_value = "123.45";
+        let precision = 5;
+        let scale = -2;
+        let err = convert_decimal_to_row_value(negative_scale_value, precision, scale).unwrap_err();
+        match err {
+            DecimalConversionError::ScaleExceeds {
+                value,
+                expected_scale,
+                actual_scale,
+            } => {
+                assert_eq!(value, negative_scale_value.to_string());
+                assert_eq!(expected_scale, scale);
+                assert_eq!(actual_scale, 2);
+            }
+            _ => panic!("Expected a ScaleExceeds error, but got a different variant: {err:?}"),
+        }
+
+        let overflow_value = "1234567890123456789012345678901234567.789";
+        let precision = 40;
+        let scale = 3;
+        let err = convert_decimal_to_row_value(overflow_value, precision, scale).unwrap_err();
+
+        match err {
+            DecimalConversionError::Overflow { mantissa, err_msg } => {
+                assert_eq!(mantissa, "1234567890123456789012345678901234567789");
+                assert!(err_msg.contains("out of range")); // Ensure the error message contains means it is out of range
+            }
+            _ => panic!("Expected an Overflow error, but got a different variant: {err:?}"),
+        }
+    }
+
+    #[test]
+    fn test_convert_decimal_to_row_value_valid() {
+        let valid_value_1 = "123.45";
+        let precision = 5;
+        let scale = 2;
+        let result = convert_decimal_to_row_value(valid_value_1, precision, scale).unwrap();
+        assert_eq!(result, RowValue::Decimal(12345));
+
+        let valid_value_2 = "123.4";
+        let precision = 5;
+        let scale = 3;
+        let result = convert_decimal_to_row_value(valid_value_2, precision, scale).unwrap();
+        assert_eq!(result, RowValue::Decimal(123400));
+
+        let valid_negative_value = "-123.4";
```

Comment:
This is a bug:
```sql
postgres=# CREATE TABLE my_table (
    my_decimal DECIMAL(5, 3)
);
CREATE TABLE
postgres=# INSERT INTO my_table (my_decimal) VALUES (-123.4);
ERROR:  numeric field overflow
DETAIL:  A field with precision 5, scale 3 must round to an absolute value less than 10^2.
```
so another "rule" for decimal is, you need to consider integer part as well

Reply:
Suggest to check with pg decimal for the correct behavior (it's installed in devcontainer already, simply `psql`)